lib/test/TestParser.js

```diff
@@ -49,10 +49,17 @@ class TestParser {
                         "Configuration element is not an object",
                         Util.extractLine(documents[0].configuration));
                 }
-                configuration = documents[0].configuration;
+                configuration = configurationOverride || documents[0].configuration;
```

Comment:
it seems that the main problem is that when you parse you overwrite the configuration if you already have one. Could we do that with only building the TestSuite object and passing it to the test runner?